src/www/ui/async/AjaxFileBrowser.php

```diff
@@ -0,0 +1,356 @@
+        self::TITLE => _("Ajax: File Browser"),
+        self::DEPENDENCIES => array("fileBrowse"),
+        self::PERMISSION => Auth::PERM_READ
+    ));
+
+    $this->uploadDao = $this->getObject('dao.upload');
+    $this->licenseDao = $this->getObject('dao.license');
+    $this->agentDao = $this->getObject('dao.agent');
+  }
+
+  /**
+   * @param Request $request
+   * @return Response
+   */
+  protected function handle(Request $request) {
+    $upload = intval($request->get("upload"));
+    $groupId = Auth::getGroupId();
+    if (!$this->uploadDao->isAccessible($upload, $groupId)) {
+      throw new \Exception("Permission Denied");
+    }
+
+    $item = intval($request->get("item"));
+    $this->uploadtree_tablename = $this->uploadDao->getUploadtreeTableName($upload);
+    $itemTreeBounds = $this->uploadDao->getItemTreeBounds($item, $this->uploadtree_tablename);
+    $left = $itemTreeBounds->getLeft();
+    if (empty($left))
+    {
+       throw new \Exception("Job unpack/adj2nest hasn't completed.");
+    }
+    
+    $scannerAgents = array_keys($this->agentNames);
+    $scanJobProxy = new ScanJobProxy($this->agentDao, $upload);
+    $scanJobProxy->createAgentStatus($scannerAgents);
+    $selectedAgentId = intval($request->get('agentId'));
+    $tag_pk = intval($request->get('tag'));
+
+    $UniqueTagArray = array();
+    $this->licenseProjector = new LicenseMap($this->getObject('db.manager'),$groupId,LicenseMap::CONCLUSION,true);
+    $vars = $this->createFileListing($tag_pk, $itemTreeBounds, $UniqueTagArray, $selectedAgentId, $groupId, $scanJobProxy);
+    
+    return new JsonResponse(array(
+            'sEcho' => intval($request->get('sEcho')),
+            'aaData' => $vars['fileData'],
+            'iTotalRecords' => intval($request->get('totalRecords')),
+            'iTotalDisplayRecords' => $vars['iTotalDisplayRecords']
+          ) );
+  }
+
+
+  /**
+   * @param $tagId
+   * @param ItemTreeBounds $itemTreeBounds
+   * @param $UniqueTagArray
+   * @param $selectedAgentId
+   * @param int $groupId
+   * @param ScanJobProxy $scanJobProxy
+   * @return array
+   */
+  private function createFileListing($tagId, ItemTreeBounds $itemTreeBounds, &$UniqueTagArray, $selectedAgentId, $groupId, $scanJobProxy)
+  {
+    if (!empty($selectedAgentId))
+    {
+      $agentName = $this->agentDao->getAgentName($selectedAgentId);
+      $selectedScanners = array($agentName=>$selectedAgentId);
+    }
+    else
+    {
+      $selectedScanners = $scanJobProxy->getLatestSuccessfulAgentIds();
+    }
+    
+    /** change the license result when selecting one version of nomos */
+    $uploadId = $itemTreeBounds->getUploadId();
+    $isFlat = isset($_GET['flatten']);
+
+    if ($isFlat)
+    {
+      $options = array(UploadTreeProxy::OPT_RANGE => $itemTreeBounds);
+    }
+    else
+    {
+      $options = array(UploadTreeProxy::OPT_REALPARENT => $itemTreeBounds->getItemId());
+    }
+    
+    $searchMap = array();
```

Comment:
$searchMap is not used
-> either add to $options or remove the code and the search field